hack/release/terraform.tf

```diff
@@ -0,0 +1,131 @@
+provider "google" {
+  project = var.google_project
+  region  = var.google_region
+  zone    = var.google_zone
+}
+
+resource "tls_private_key" "ssh" {
+  algorithm = "RSA"
+}
+
+resource "google_compute_instance" "vm_instance" {
+  name         = "${var.prefix}-calico-release-executor"
+  machine_type = var.machine_type
+  zone         = var.google_zone
+
+  service_account {
+    scopes = ["storage-ro", "cloud-platform", "compute-rw", "logging-write", "monitoring", "service-control", "service-management"]
+  }
+
+  tags = ["calico-release"]
+
+  metadata = {
+    ssh-keys = "ubuntu:${tls_private_key.ssh.public_key_openssh}"
+  }
+
+  connection {
+    user        = "ubuntu"
+    agent       = false
+    private_key = tls_private_key.ssh.private_key_pem
+    host        = self.network_interface.0.access_config.0.nat_ip
+  }
+
+  boot_disk {
+    initialize_params {
+      image = var.image
+      type  = var.disk_type
+      size  = var.disk_size
+    }
+  }
+
+  network_interface {
+    network = var.google_network
+
+    access_config {
+      // This empty field requests an Ephemeral IP
+    }
+  }
+}
+
+resource "null_resource" "configure_vm" {
+  connection {
+    user        = "ubuntu"
+    agent       = false
+    private_key = tls_private_key.ssh.private_key_pem
+    host        = google_compute_instance.vm_instance.network_interface.0.access_config.0.nat_ip
+  }
+
+  // Install the necessary environment for release.
+  provisioner "remote-exec" {
+    inline = [
+      "sudo apt-get update",
+      "sudo apt-get install -y git make ca-certificates curl gnupg lsb-release",
+      "curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo gpg --dearmor -o /usr/share/keyrings/docker-archive-keyring.gpg",
+      "echo 'deb [arch=amd64 signed-by=/usr/share/keyrings/docker-archive-keyring.gpg] https://download.docker.com/linux/ubuntu bionic stable' | sudo tee /etc/apt/sources.list.d/docker.list > /dev/null",
+      "sudo apt-get update",
+      "sudo apt-get install -y docker-ce docker-ce-cli containerd.io",
+      "sudo usermod -aG docker ubuntu"
+    ]
+  }
+
+  // Authenticate the docker daemon with GCR.
+  // TODO: Also handle quay and dockerhub.
+  provisioner "file" {
+    source      = var.gcr_auth_path
+    destination = "/tmp/gcr-credentials.json"
+  }
+  provisioner "remote-exec" {
+    inline = [
```

Comment:
Right now this code only logs into GCR - need to add dockerhub and quay support.